### PR TITLE
Game state changes

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Objective
- Avoid `Game.GameState` redundancy

### Changelog
- added Game.inState(state)
- Game.getGameState() changed to Game.getState()
- Game.setGameState(state) changed to Game.setState(state)

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/game.h [hunk 1]
```diff
@@ -419,16 +419,20 @@ class Game
 	void updateCreatureWalkthrough(const Creature* creature);
 	void updateKnownCreature(const Creature* creature);
 
-	GameState_t getGameState() const;
-	void setGameState(GameState_t newState);
-	void saveGameState();
+	template <GameState_t S>
```
Comment (MillhioreBT) [inline src/game.h:422]: there is no point in using a template in this case, we should avoid that if not necessary

**Review by MillhioreBT — CHANGES_REQUESTED** [review]

File: src/game.h (hunk 1 above)
Comment (MillhioreBT) [reply src/game.h:422]: just use a classic argument of type GameState_t and that's it, why add so much magic for a simple return int == int;

**Review by EPuncker — APPROVED** [review]

File: data/lib/compat/compat.lua [hunk 2]
```diff
@@ -1350,6 +1350,18 @@ function doSetGameState(state)
 	return Game.setGameState(state)
 end
 
+function Game.inGameState(state)
+	return Game.inState(state)
```
Comment (ranisalt) [inline data/lib/compat/compat.lua:1354]:
```suggestion
	return Game.getState() == state
```

Drop that C++ thingy

File: src/game.h [hunk 3]
```diff
@@ -419,16 +419,19 @@ class Game
 	void updateCreatureWalkthrough(const Creature* creature);
 	void updateKnownCreature(const Creature* creature);
 
-	GameState_t getGameState() const;
-	void setGameState(GameState_t newState);
-	void saveGameState();
+	bool inState(GameState_t state) const
+	{
+		return gameState == state;
+	}
```
Comment (ranisalt) [inline src/game.h:425]: Same, you can drop it and move the definition of `getState` to the header so it gets inlined (likely gets inlined anyway with LTO)

**Comment by nekiro** [comment]
What's the benefit from this change? I fail to find one, looks purely cosmetical

**Comment by ramon-bernardo** [comment]
> What's the benefit from this change? I fail to find one, looks purely cosmetical and irrelevant

Yeah, i like cosmetical code

**Pull request closed without merging**